battlepass/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+			ensure!(creator == prime, Error::<T>::AuthorizationError);  												// TODO: Add Root temporaryly
+			// check is there is no active battlepass for the Org
+			ensure!(!ActiveBattlepassByOrg::<T>::contains_key(&org_id), Error::<T>::BattlepassExists);
+			let new_season = Self::get_battlepass_count(org_id) + 1;
+			let now = <frame_system::Pallet<T>>::block_number();
+
+			// Create collection 
+			let metadata = BoundedVec::truncate_from(b"meta".to_vec());		// TODO: what should be here?
+			let symbol = BoundedVec::truncate_from(b"symbol".to_vec());		// TODO: what should be here?
+			let collection_id = T::Rmrk::collection_create(creator.clone(), metadata, None, symbol)?;
+
+			// Create Battlepass
+			let battlepass: Battlepass<T::Hash, T::AccountId, T::BlockNumber, String<T>> = types::Battlepass {
+				creator, 
+				org_id,
+				name,
+				cid,
+				season: new_season,
+				collection_id,
+				price,
+				created: now.clone(), mutated: now
+			};
+			let battlepass_id = <T as frame_system::Config>::Hashing::hash_of(&battlepass);
+
+			Battlepasses::<T>::insert(&battlepass_id, battlepass);
+			BattlepassStates::<T>::insert(&battlepass_id, types::BattlepassState::Active); 								// TODO: change to Draft
+			ActiveBattlepassByOrg::<T>::insert(org_id, battlepass_id);
+			BattlepassCount::<T>::insert(org_id, new_season);
+
+			Self::deposit_event(Event::BattlepassCreated { org_id, battlepass_id, season: new_season });
+
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		pub fn claim_battlepass(
+			origin: OriginFor<T>,
+			org_id: T::Hash,
+		) -> DispatchResult {
+			let claimer = ensure_signed(origin)?;
+			// check if Org exists
+			ensure!(T::Control::is_org_active(&org_id), Error::<T>::OrgUnknown);
+			// check if user is a member of organization
+			ensure!(T::Control::is_org_member_active(&org_id, &claimer), Error::<T>::NotMember);
+			// check if Org has active battlepass
+			let battlepass_id = Self::get_active_battlepass(org_id).ok_or(Error::<T>::BattlepassUnknown)?;
+			// check if Battlepass already claimed
+			ensure!(!ClaimedBattlepass::<T>::contains_key(claimer.clone(), battlepass_id), Error::<T>::BattlepassClaimed);
+
+			let battlepass = Self::get_battlepass(battlepass_id).ok_or(Error::<T>::BattlepassUnknown)?;
+			let collection = <pallet_rmrk_core::Pallet<T>>::collections(battlepass.collection_id).ok_or(Error::<T>::CollectionUnknown)?;
+
+			// Create NFT
+			let (collection_id, nft_id) = T::Rmrk::nft_mint(
+				claimer.clone(), 									// sender
+				claimer.clone(), 									// owner
+				collection.nfts_count, 								// nft_id
+				battlepass.collection_id, 							// collection_id
+				None, 												// royalty_recipient
+				None, 												// royalty_amount
+				BoundedVec::truncate_from(b"NFT meta".to_vec()), 	// metadata 			TODO: what should be here?
+				false, 												// transferable
+				None												// resources
```

Comment:
I don't like these inline comments.
Here are some [standards for documentation of the Rust project](https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#appendix-a-full-conventions-text).